src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -156,8 +156,11 @@ private static void setAcl(Session session, List<String> principals, String jcrP
             if (principal == null) {
                 // backwards compatibility: fallback to original code treating principal name as authorizable ID (see SLING-8604)
                 final Authorizable authorizable = UserUtil.getAuthorizable(session, name);
-                checkState(authorizable != null, "Authorizable not found: {0}", name);
-                principal = authorizable.getPrincipal();
+                if (authorizable != null) {
+                    principal = authorizable.getPrincipal();
+                } else {
+                    principal = () -> name;
```

Comment:
since the previous behavior was to throw an exception, i would recommend to at least log a warning here.

Reply:
I think this is resolved with my latest changes.

Reply:
yes, i agree